libapicore/ApiServer.h

```diff
@@ -58,16 +64,19 @@ class ApiConnection
 	Json::FastWriter m_jWriter;
 
 	bool m_readonly =  false ;
+	std::string m_password = "";
 	Farm& m_farm;
 
+	bool m_is_authenticated = true;
```

Comment:
Can we always init that to `false` but require authentication only if the password is not empty?

Reply:
And in fact it is like that. It's m_is_authenticated, not m_requires_auth